src/AspNet.Security.OAuth.Apple/AppleAuthenticationHandler.cs

```diff
@@ -144,7 +138,8 @@ protected virtual IEnumerable<Claim> ExtractClaimsFromToken([NotNull] string tok
         {
             try
             {
-                var securityToken = _tokenHandler.ReadJwtToken(token);
+                var tokenHandler = Options.JwtSecurityTokenHandler ?? new System.IdentityModel.Tokens.Jwt.JwtSecurityTokenHandler();
```

Comment:
Is it worth having a null check here?

Also, it might be the right time to move to the newer `JsonWebTokenHandler` stack.

Reply:
I'm not a big fan of making the code assume validation elsewhere too much, as if it's wrong/changed later then you get harder-to-diagnose `NullReferenceException`s instead of it either "just working", or giving you a more useful error.

I was just changing it locally to assign the default as a `PostConfigure()` and remove the null checks, but that then broke some of the unit tests that don't go through the full OAuth pipeline as the post-configure code doesn't run, so they just throw `NullReferenceException` rather than work, which sort of fits with why I'm not a fan of not-checking it in the first place.

Reply:
Ignoring the JWT handler from the options if it cannot be found and falling back to a default instance here - which is hard to detect - seems weird to me. Why not throwing an exception instead?

Regarding the tests, it's not clear to me why you'd get a null-ref: the property is always assigned a default value using the `{ get; set; } = new ...` syntax.

Reply:
Not in the commit I pushed just now where I copied the `= default!;` pattern from the AspNetCore repo which you hadn't seen where it's initialised by a `PostConfigure()` call.

But I fixed it by just explicitly setting the JwtSecurityTokenHandler property in the 3 tests it affected.

Reply:
I refactored the tests so that the `PostConfigure()` got called which fixed the NREs.

I looked at changing over to `JsonWebTokenHandler`, but that broke all the tests as there's no longer a hook the tests can use to override the lifetime validation of the JWT without adding additional functionality to allow the application to configure the token validation parameters.

I'll leave that for now as it's a bigger change and unrelated to the original issue, plus would need the SuperOffice provider to change as well as that uses the same handler for JWTs.